ecs-init/docker/docker.go

```diff
@@ -246,10 +251,20 @@ func (c *Client) getContainerConfig() *godocker.Config {
 	}
 }
 
-func (c *Client) loadEnvVariables() map[string]string {
+// LoadEnvVariables gets user-supplied environment variables
+func (c *Client) LoadEnvVariables() map[string]string {
```

Comment:
Why are these methods on the Client when these are products entirely of configuration? Is this here only for testing?

Reply:
this method is already present today. the method felt out of place, but I tried to use it as is and avoid too much refactoring work.

Reply:
The current usage doesn't depend on the `*Client` here, this could be moved to the config package as a function there and have the Client hold or accept a reference to a `var environment map[string]string` for when it requires it. There shouldn't be need for consumers of the configured environment to have to know how to merge the variables.

I feel most strongly about having the merge logic hidden and managed centrally, I don't think we *must* have the implementation moved. It is certainly desirable though.

Reply:
The existing design is that `Client` has filesystem embedded, which it uses to read the config files.
I agree this is not the best way to do this, but to change this, it will require moving the filesystem out to a different package like utils. 
I can add a `TODO` and do the refactoring work in a different PR. 

With regards to hiding the merging logic, `LoadEnvVariables()`  and `LoadCustomInstanceEnvVars()` can be combined into a single method that engine can use. Let me know what you think.